### PR TITLE
Add fill to select statements.

### LINKED ISSUE #1913: Wire up fill options
Add support for will options: `fill(<number>)`, `fill(none)`, `fill(previous)`

### PR DESCRIPTION
**Pull request by pauldix**
Fixes #1913
Add `fill(none)`, `fill(<num>)`, and `fill(previous)` options for replacing null values in aggregates. The none option will remove any rows from values if there is a null value anywhere in there.

### CODE REVIEW

File: influxql/ast.go [hunk 1]
```diff
@@ -529,6 +529,19 @@ func (s *AlterRetentionPolicyStatement) RequiredPrivileges() ExecutionPrivileges
...
 }
 
+type FillOption int
+
+const (
+	// NullFill means that empty aggregate windows will just have null values.
+	NullFill FillOption = iota
+	// NoFill means that empty aggregate windows will be purged from the result.
+	NoFill
+	// NumberFill means that empty aggregate windwos will be filled with the given number
```
Comment (otoolep) [inline influxql/ast.go]: Nit: typo in comment.

**Comment by otoolep** [comment]
LGTM, nitpick feedback.

**Comment by paaland** [comment]
fill(previous) works only on the current time span (the where statement). I expected it to use the previously logged value at or before the current time span.

Say I have the value 100 logged at 2014-12-31
And the value 50 logged at 2015-01-02

The I perform the query:

select value from someSeries where time > '2015-01-01' group by time(24h) fill(previous)

It returns
2015-01-01 null
2015-01-02 50
2015-01-03 50
2015-01-04 50
and so one

I would expect it to return 100 as the value for 2015-01-01 when specifying fill(previous).
In other historian db's you have the notion of "include bounds" which will include the last value before as well as the first value after your specified time range. These can be used to fill in manually.

**Comment by beckettsean** [comment]
@paaland that's a valuable insight and a reasonable feature request. Can you raise it as a new issue on GitHub rather than a comment on this merged pull request?

**Pull request merged**